From: Takashi Iwai <tiwai@suse.de>
Subject: Workaround for mysterious NVMe breakage with i915 CFL
Patch-mainline: Never, a temporary workaround
References: bsc#1111040

A temporary workaround for a new CFL that breaks NVMe by some reason
at detecting the non-existing HDMI-3 port.

This should be removed once when it gets fixed properly in the upstream
code.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/gpu/drm/i915/intel_bios.c |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/gpu/drm/i915/intel_bios.c
+++ b/drivers/gpu/drm/i915/intel_bios.c
@@ -2031,11 +2031,17 @@ intel_bios_is_lspcon_present(struct drm_
 				enum port port)
 {
 	const struct child_device_config *child;
+	struct pci_dev *pdev = dev_priv->drm.pdev;
 	int i;
 
 	if (!HAS_LSPCON(dev_priv))
 		return false;
 
+	// XXX: workaround for a buggy Intel board (bsc#1111040)
+	if (pdev->subsystem_vendor == 0x8086 &&
+	    pdev->subsystem_device == 0x2212)
+		return true;
+
 	for (i = 0; i < dev_priv->vbt.child_dev_num; i++) {
 		child = dev_priv->vbt.child_dev + i;
 
